### PR TITLE
[DO NOT MERGE] Standalone Beats OTel collector PoC

### PR DESCRIPTION
**Pull request by rdner**
This is the results of our experiment embedding OTel collector into Beats for running them in the standalone mode. 

-- 

Steps to run filebeat in otel mode:

1. Clone from this branch.
2. Build the `filebeat` binary with `mage package`
3. Run the binary using
```sh
./filebeat otel --config /path/to/otel/config
```

The filebeat-otel mode makes use of `filebeatreceiver` and config of that same as that of standalone filebeat. 

--- 
Currently, we are using hardcoded OpenTelemetry configurations, as the conversion from Filebeat to OpenTelemetry is nearly complete. There are still a few remaining tasks to address, and we’re actively working on them.

### CODE REVIEW

**Comment by VihasMakwana** [comment]
Attaching a demo of docker autodiscovery using `filebeat` (standalone otel mode)

https://github.com/user-attachments/assets/f513f4d2-74a9-4d61-8e6b-907a85b4eb4d

cc: @cmacknz

**Comment by belimawr** [comment]
Folks, I've been trying to test it but I could not get my ES config translation function to work/be called, so I made some changes, now it is used, however as it is it's first test, there are some issues I haven't fixed yet.

Currently I'm using a hack to get the ES output configuration and use it as OTel ES exporter:
1. In `cmd/otel.yml` I added an `output` section with the Beats ES output configuration, that's the one parsed by https://github.com/elastic/beats/blob/216cf68e56969bd601aae638af48082a739dfef6/x-pack/filebeat/cmd/customProvider/beatprovider.go#L43
2. For some reason skipping TLS validation is not working (one of the many things to fix), so I'm using a Cloud deployment.

The way I'm testing/running:
```
cd x-pack/filebeat
go build .
./filebeat otel -e -v -d "*" -config filebeat:./cmd/otel.yml
```

I can see the config is correctly parsed, converted and used to create an Elasticsearch exporter, however the index seems to be missing. If you let it running for a while you'll get an error from the exporter.

**Comment by belimawr** [comment]
I finally managed to get all the different bits working together :tada: 

TL;DR:
Disabling TLS is not working, it might be an issue in the Elasticsearch exporter, so just make sure to either not use TLS or use an ES/Kibana with valid certificates. If you want the dataview to be automatically created, you need to setup the Kibana host and credentials
The configuration file for the demo is `x-pack/filebeat/cmd/filebeat.yml`
```
cd x-pack/filebeat
# This will setup ILM, Index templates, data view, etc
go run . setup -e -v
# Actually run Filebeat as OTel collector using the converted ES output settings
go run . otel -e -v -d "*" -config filebeat:./cmd/filebeat.yml
```

**Comment by rdner** [comment]
This PoC got polished and finalized in https://github.com/elastic/beats/pull/41849
Closing.
